engine/pom.xml

```diff
@@ -333,6 +333,10 @@
       <version>${project.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
+    </dependency>
```

Comment:
Lmk if we prefer not to introduce this dependency for some reason, can also just do the `StringUtils.isNotEmpty` check manually.

Reply:
Cool thing will adjust and remove the dependency